Common/Extensions.cs

```diff
@@ -2505,11 +2505,22 @@ public static bool TryConvert<T>(this PyObject pyObject, out T result, bool allo
             {
                 try
                 {
+                    if (allowPythonDerivative)
+                    {
+                        result = (T)pyObject.AsManagedObject(type);
```

Comment:
Let's add a comment of **why** is this here. 
Would be awesome to expand the tests around `TryConvert` for the specific issues solved and handled here